pkg/serverless/appsec/httpsec/http.go

```diff
@@ -65,24 +73,101 @@ func parseBody(headers map[string][]string, rawBody *string) (body interface{})
 	}
 	raw := []byte(*rawBody)
 
-	var ct string
-	if values, ok := headers["content-type"]; !ok {
+	mimeHeaders := make(textproto.MIMEHeader, len(headers))
+	for key, values := range headers {
+		for _, value := range values {
+			mimeHeaders.Add(key, value)
+		}
+	}
+
+	result, err := tryParseBodyBytes(mimeHeaders, raw)
+	if err != nil {
+		log.Warnf("unable to parse request body: %v", err)
 		return rawStr
-	} else if len(values) > 1 {
+	} else if result == nil {
 		return rawStr
+	}
+
+	return result
+}
+
+// / tryParseBodyBytes attempts to parse the raw data in `raw` according to the `headers`. Returns an error if parsing
+// / fails, and a `nil` body if no parsing strategy was found.
+func tryParseBodyBytes(headers textproto.MIMEHeader, raw []byte) (body interface{}, err error) {
```

Comment:
Why not - I was getting the `[]byte` in there because the multipart segments may contain arbitrary binary data and I find it a little gross to wrap that into a string... That's now changed in the PR :)

Reply:
The reason why it's worth it is because the top-level source is a string in the parsed invocation request, so passing it as a string down the way saves some copies.